sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java

```diff
@@ -1,42 +1,102 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-
 package com.azure.cosmos.models;
 
+import com.azure.cosmos.BridgeInternal;
 import com.azure.cosmos.CosmosDiagnostics;
 import com.azure.cosmos.implementation.CosmosItemProperties;
+import com.azure.cosmos.implementation.Document;
+import com.azure.cosmos.implementation.ResourceResponse;
+import com.azure.cosmos.implementation.SerializationDiagnosticsContext;
+import com.azure.cosmos.implementation.Utils;
 
 import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
 import java.util.Map;
 
 /**
- * The synchronous Cosmos item response. Contains methods to access the Item and other response methods
+ * The type Cosmos item response. This contains the item and response methods
  *
  * @param <T> the type parameter
  */
 public class CosmosItemResponse<T> {
-    private final CosmosAsyncItemResponse<T> responseWrapper;
-
-    CosmosItemResponse(CosmosAsyncItemResponse<T> response) {
-        this.responseWrapper = response;
+    private final Class<T> itemClassType;
+    private final byte[] responseBodyAsByteArray;
+    private T item;
+    private final ResourceResponse<Document> resourceResponse;
+    private CosmosItemProperties props;
+
+    CosmosItemResponse(ResourceResponse<Document> response, Class<T> classType) {
+        this.itemClassType = classType;
+        this.responseBodyAsByteArray = response.getBodyAsByteArray();
+        this.resourceResponse = response;
     }
 
     /**
-     * Gets resource.
+     * Gets the resource .
```

Comment:
Typo: space between “resource” and period.

Reply:
done